src/quartz_api/internal/service/sites.py

```diff
@@ -128,4 +129,12 @@ def post_generation(
     # get email from auth
     email = auth["https://openclimatefix.org/email"]
 
+    # check for nans
+    for entry in generation:
+        if np.isnan(entry.PowerKW):
+            raise HTTPException(
+                status_code=status.HTTP_400_BAD_REQUEST,
+                detail="PowerKW values must be floating point numbers (no NaNs allowed).",
```

Comment:
Technically, depending on the floating point specification, NaN is a valid floating point number - but I think that isn't going to be a cause of confusion here! I'm not sure the comment is necessary though - it's fairly obvious what's going on here.